core/src/createTextMaskInputElement.js

```diff
@@ -9,14 +9,62 @@ const strNone = 'none'
 
 export default function createTextMaskInputElement({
   inputElement,
-  mask: providedMask,
+  mask,
   guide,
-  pipe,
+  pipe: providedPipe,
   placeholderChar = defaultPlaceholderChar,
   onAccept,
   onReject,
   keepCharPositions = false
 }) {
+  // Anything that we will need to keep between `update` calls, we will store in this `state` object.
+  const state = {previousConformedValue: emptyString}
+
+  const {
+    mask: providedMask,
+    placeholder,
+    pipe
+  } = processMaskAndPlaceholder({
+    mask,
+    pipe: providedPipe,
+    placeholderChar
+  })
+
+  return {
+    state,
+    // `update` is called by framework components whenever they want to update the `value` of the input element.
+    // The caller can send a `rawValue` to be conformed and set on the input element. However, the default use-case
+    // is for this to be read from the `inputElement` directly.
+    update: function(rawValue = inputElement.value) {
+      // `selectionStart` indicates to us where the caret position is after the user has typed into the input
+      const {selectionStart: currentCaretPosition} = inputElement
+
+      const {value = '', adjustedCaretPosition} = getConformedInputState({
```

Comment:
Do you need to default `value` to an empty string? If that's necessary, use `emptyString` variable.